build_tools/aws-sdk-code-generator/templates/types_module.mustache

```diff
@@ -18,6 +18,9 @@ module {{module_name}}
       {{/members}}
       SENSITIVE = {{sensitive_params}}
       include Aws::Structure
+      {{#union?}}
+      include Aws::Structure::Union
```

Comment:
Is there a solution that exists without a union module?

Reply:
The Union structure is being used for two things: 1) to provide some methods and 2) for "type" checking.  It could be replaced by code-genning that information directly.  eg:

```
{{#union?}}
      def member
        self.members.select { |k| self[k] }.first
      end

      def value
        self[member] if member
      end

     def union?
       true
     end
{{^union?}}
    def union?
      false
    end
{{#/union?}}
```